src/components/interactive-builder/modals/question/question-form/question/question.component.tsx

```diff
@@ -31,6 +32,15 @@ const Question: React.FC<QuestionProps> = ({ checkIfQuestionIdExists }) => {
     return checkIfQuestionIdExists(formField.id);
   }, [formField.id, checkIfQuestionIdExists]);
 
+  const handleQuestionTypeChange = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>) => {
+      const newType = event.target.value;
+      const cleaned = cleanFormFieldForType({ ...formField, type: newType }, newType);
+      setFormField(cleaned);
+    },
+    [formField, setFormField],
```

Comment:
Having formField and setFormField as dependencies aren't good because setFormField updated the formField. Check the other functions to see a better way to use setFormField - https://github.com/openmrs/openmrs-esm-form-builder/blob/eff9e72db3964140f0c1068d9086397a45236520/src/components/interactive-builder/modals/question/question-form/question-types/inputs/obs/obs-type-question.component.tsx#L32